phpBB/includes/acp/acp_extensions.php

```diff
@@ -382,11 +422,350 @@ function main()
 	}
 
 	/**
-	* Lists all the enabled extensions and dumps to the template
-	*
-	* @return null
-	*/
-	public function list_enabled_exts()
+	 * Handles the catalog mode of the extensions list
+	 *
+	 * @param string $id
+	 * @param string $mode
+	 */
+	public function catalog_mode($id, $mode)
+	{
+		global $phpbb_container;
+
+		$action = $this->request->variable('action', 'list');
+
+		/** @var \phpbb\language\language $language */
+		$language = $phpbb_container->get('language');
+
+		/** @var \phpbb\composer\manager $composer_manager */
+		$composer_manager = $phpbb_container->get('ext.composer.manager');
+
+		/** @var \phpbb\extension\manager $extensions_manager */
+		$extensions_manager = $phpbb_container->get('ext.manager');
+
+		if (!$composer_manager->check_requirements())
+		{
+			$this->page_title = 'ACP_EXTENSIONS_CATALOG';
+			$this->tpl_name = 'message_body';
+
+			$this->template->assign_vars([
+				'MESSAGE_TITLE'	=> $language->lang('EXTENSIONS_CATALOG_NOT_AVAILABLE'),
+				'MESSAGE_TEXT'	=> $language->lang('EXTENSIONS_COMPOSER_NOT_WRITABLE'),
+			]);
+
+			return;
+		}
+
+		switch ($action)
+		{
+			case 'install':
+				$this->page_title = 'ACP_EXTENSIONS_INSTALL';
+
+				$extension = $this->request->variable('extension', '');
+
+				if (empty($extension))
+				{
+					redirect($this->u_action);
+				}
+
+				$formatter = new \phpbb\composer\io\html_output_formatter([
+					'warning' => new \Symfony\Component\Console\Formatter\OutputFormatterStyle('black', 'yellow')
+				]);
+
+				$composer_io = new \phpbb\composer\io\web_io($language, '', $phpbb_container->getParameter('extensions.composer.output'), $formatter);
+
+				try
+				{
+					$this->request->enable_super_globals();
```

Comment:
Actually it is a requirement or composer and I'm wondering if it should be moved to composer/installer